api/jobs.go

```diff
@@ -119,12 +120,51 @@ func (api *JobStoreAPI) GetJobsHandler(w http.ResponseWriter, req *http.Request)
 		http.Error(w, serverErrorMessage, http.StatusInternalServerError)
 		return
 	}
-
 }
 
 // unlockJob unlocks the provided job lockID and logs any error with WARN state
 func (api *JobStoreAPI) unlockJob(ctx context.Context, lockID string) {
+	log.Event(ctx, "Entering unlockJob function, which unlocks the provided job lockID.", log.INFO)
 	if err := api.jobStore.UnlockJob(lockID); err != nil {
 		log.Event(ctx, "error unlocking lockID for a job resource", log.WARN, log.Data{"lockID": lockID})
 	}
 }
+
+// PutNumTasksHandler returns a function that updates the number_of_tasks in an existing Job resource, which is associated with the id passed in.
+func (api *JobStoreAPI) PutNumTasksHandler(ctx context.Context) http.HandlerFunc {
+	log.Event(ctx, "Creating handler function, which calls PutNumberOfTasks to update the number_of_tasks in the job with the supplied id.", log.INFO)
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		vars := mux.Vars(req)
+		id := vars["id"]
+		count := vars["count"]
+
+		var numTasks int
+		numTasks, err := strconv.Atoi(count)
+		logData := log.Data{"job_id": id, "num_tasks": numTasks}
```

Comment:
Minor point - would it be more beneficial to do 
```
log.Data{"job_id": id, "num_tasks": count}
```
If count is not a valid int or cannot be parsed to an int then `numTasks` will be the zero value for it so it will hide what the offending value was. If you add count instead you will be able to see what the value was which might be useful if this request fails with an error. As I said really minor point but could be really helpful in the right circumstances.

Reply:
I agree this could be useful, I had forgotten that I could put an int in the log data, thanks for the idea.